4-domain.go

```diff
@@ -0,0 +1,109 @@
+package domain
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
+	"github.com/nspcc-dev/neo-go/pkg/interop/storage"
+)
+
+// Main is a very useful function.
+func Main(operation string, args []interface{}) interface{} {
+	ctx := storage.GetContext()
+
+	// Queries the domain owner
+	if operation == "query" {
+		if checkArgs(args, 1) {
+			domainName := args[0].([]byte)
+			message := "QueryDomain: " + args[0].(string)
+			runtime.Notify(message)
+
+			owner := storage.Get(ctx, domainName)
+			if owner != nil {
+				runtime.Notify(owner)
+				return owner
+			}
+
+			runtime.Notify("Domain is not yet registered")
+		}
+	}
+
+	// Deletes the domain
+	if operation == "delete" {
+		if checkArgs(args, 1) {
+			domainName := args[0].([]byte)
+			message := "DeleteDomain: " + args[0].(string)
+			runtime.Notify(message)
+
+			owner := storage.Get(ctx, domainName)
+			if owner == nil {
+				runtime.Notify("Domain is not yet registered")
+				return false
+			}
+
+			if !runtime.CheckWitness(owner.([]byte)) {
+				runtime.Notify("Sender is not the owner, cannot delete")
+				return false
+			}
+
```

Comment:
And here (and everywhere else we change the state) make a structured notification (that could be seen in the application log).